src/handlers/flow/FlowStreamHandler.h

```diff
@@ -0,0 +1,222 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at https://mozilla.org/MPL/2.0/. */
+
+#pragma once
+
+#include "AbstractMetricsManager.h"
+#include "FlowInputStream.h"
+#include "MockInputStream.h"
+#include "StreamHandler.h"
+#include <Corrade/Utility/Debug.h>
+#include <IPv4Layer.h>
+#include <IPv6Layer.h>
+#include <string>
+
+namespace visor::handler::flow {
+
+using namespace visor::input::mock;
+using namespace visor::input::flow;
+
+typedef std::pair<in_addr, uint8_t> Ipv4Subnet;
+typedef std::pair<in6_addr, uint8_t> Ipv6Subnet;
+
+namespace group {
+enum FlowMetrics : visor::MetricGroupIntType {
+    Counters,
+    Cardinality,
+    TopGeo,
+    TopByPackets,
+    TopByBytes
+};
+}
+
+struct FlowData {
+    bool is_ipv6;
+    IP_PROTOCOL l4;
+    size_t payload_size;
+    uint32_t packets;
+    pcpp::IPv4Address ipv4_in;
+    pcpp::IPv4Address ipv4_out;
+    pcpp::IPv6Address ipv6_in;
+    pcpp::IPv6Address ipv6_out;
+    uint16_t src_port;
+    uint16_t dst_port;
+    uint32_t if_in_index;
+    uint32_t if_out_index;
+};
+
+struct FlowPacket {
+    timespec stamp;
+    std::vector<FlowData> flow_data;
+
+    FlowPacket(timespec stamp)
+        : stamp(stamp)
+    {
+    }
+};
+
+class FlowMetricsBucket final : public visor::AbstractMetricsBucket
+{
+
+protected:
+    mutable std::shared_mutex _mutex;
+
+    Cardinality _srcIPCard;
+    Cardinality _dstIPCard;
+
+    TopN<std::string> _topGeoLoc;
+    TopN<std::string> _topASN;
+
+    struct topns {
+        TopN<std::string> topSrcIP;
+        TopN<std::string> topDstIP;
+        TopN<uint16_t> topSrcPort;
+        TopN<uint16_t> topDstPort;
+        TopN<uint32_t> topInIfIndex;
+        TopN<uint32_t> topOutIfIndex;
+        topns(std::string metric)
+            : topSrcIP("flows", "ip", {"top_src_ip_" + metric}, "Top source IP addresses by " + metric)
+            , topDstIP("flows", "ip", {"top_dst_ip_" + metric}, "Top destination IP addresses by " + metric)
+            , topSrcPort("flows", "port", {"top_src_ports_" + metric}, "Top source ports by " + metric)
+            , topDstPort("flows", "port", {"top_dst_ports_" + metric}, "Top destination ports by " + metric)
+            , topInIfIndex("flows", "index", {"top_in_if_index_" + metric}, "Top input interface indexes  by " + metric)
+            , topOutIfIndex("flows", "index", {"top_out_if_index_" + metric}, "Top output interface indexes by " + metric)
+        {
+        }
+    };
+
+    topns _topByBytes;
+    topns _topByPackets;
+
+    // total numPackets is tracked in base class num_events
```

Comment:
i think this comment doesn't apply, right? or maybe it's numFlowPackets essentially?